pkg/scheduler/schedule_one_test.go

```diff
@@ -2050,6 +2052,178 @@ func TestSchedulerSchedulePod(t *testing.T) {
+// and it doesn't check the response from the function carefully.
+// The behavior should be validated in the TestSchedulerSchedulePod test.
+func BenchmarkSchedulePod(b *testing.B) {
+	preFilterPlugins := make([]st.RegisterPluginFunc, 20)
+	for i := 0; i < 20; i++ {
+		pluginName := "FakePreFilter" + strconv.Itoa(i)
+		preFilterPlugins[i] = st.RegisterPreFilterPlugin(
+			pluginName,
+			st.NewFakePreFilterPlugin(pluginName, nil, nil),
+		)
+	}
+	filterPlugins := make([]st.RegisterPluginFunc, 20)
+	for i := 0; i < 20; i++ {
+		pluginName := "FakeFilter" + strconv.Itoa(i)
+		filterPlugins[i] = st.RegisterFilterPlugin(pluginName, st.NewTrueFilterPlugin)
+	}
+	preScorePlugins := make([]st.RegisterPluginFunc, 20)
+	for i := 0; i < 20; i++ {
+		pluginName := "FakePreScore" + strconv.Itoa(i)
+		preScorePlugins[i] = st.RegisterPreScorePlugin(pluginName, st.NewFakePreScorePlugin(pluginName, nil, nil))
+	}
+	scorePlugins := make([]st.RegisterPluginFunc, 20)
+	for i := 0; i < 20; i++ {
+		pluginName := "FakeScore" + strconv.Itoa(i)
+		scorePlugins[i] = st.RegisterScorePlugin(pluginName, newNumericMapPlugin(pluginName), 1)
+	}
+
+	extenders := make([]framework.Extender, 20)
+	for i := 0; i < 20; i++ {
+		extenderName := "FakeExtender" + strconv.Itoa(i)
+		extenders[i] = &st.FakeExtender{
+			ExtenderName: extenderName,
+			Predicates:   []st.FitPredicate{st.TruePredicateExtender},
+			Prioritizers: []st.PriorityConfig{{Function: st.Node1PrioritizerExtender, Weight: 10}},
+			Weight:       1,
+		}
+	}
+
+	allPlugins := append(append(append(preFilterPlugins, filterPlugins...), scorePlugins...), preScorePlugins...)
+
+	nodes := make([]string, 100)
+	for i := 0; i < 100; i++ {
+		nodes[i] = strconv.Itoa(i)
+	}
+	tests := []struct {
+		name            string
+		registerPlugins []st.RegisterPluginFunc
+		extenders       []framework.Extender
+		nodes           []string
+		pvcs            []v1.PersistentVolumeClaim
+		pod             *v1.Pod
+		pods            []*v1.Pod
+	}{
+		{
+			name: "Scheduler has some filter plugins, but has no other plugins.",
+			registerPlugins: append([]st.RegisterPluginFunc{
+				st.RegisterQueueSortPlugin(queuesort.Name, queuesort.New),
+				st.RegisterBindPlugin(defaultbinder.Name, defaultbinder.New),
+			}, filterPlugins...),
+			nodes: nodes,
+			pod:   st.MakePod().Name("pod").Obj(),
+		},
+		{
+			name: "Scheduler has some score plugins, but has no other plugins.",
+			registerPlugins: append([]st.RegisterPluginFunc{
+				st.RegisterQueueSortPlugin(queuesort.Name, queuesort.New),
+				st.RegisterBindPlugin(defaultbinder.Name, defaultbinder.New),
+			}, scorePlugins...),
+			nodes: nodes,
+			pod:   st.MakePod().Name("pod").Obj(),
+		},
+		{
+			name: "Scheduler has some pre filter plugins, but has no other plugins.",
+			registerPlugins: append([]st.RegisterPluginFunc{
+				st.RegisterQueueSortPlugin(queuesort.Name, queuesort.New),
+				st.RegisterBindPlugin(defaultbinder.Name, defaultbinder.New),
```

Comment:
The below code could be split out, so you won't duplicate them in all of those testcases, 
```
			registerPlugins: append([]st.RegisterPluginFunc{
				st.RegisterQueueSortPlugin(queuesort.Name, queuesort.New),
				st.RegisterBindPlugin(defaultbinder.Name, defaultbinder.New),
```